framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,262 @@
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ETimporter, cls).getInputSpecification()
+
+    return inputSpecification
+
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+    self.inputs = inputs
+    self._workingDir = runInfo['WorkingDir']
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    for child in xmlNode:
+      if child.tag == 'fileFormat':
+        if child.text not in self.allowedFormats:
+          self.raiseAnError(IOError, 'ETimporterPostProcessor Post-Processor ' + self.name + ', format ' + child.text + ' : is not supported')
+        else:
+          self.ETformat = child.text
+      else:
+        self.raiseAnError(IOError, 'ETimporterPostProcessor Post-Processor ' + self.name + ', node ' + child.tag + ' : is not recognized')
+
+  def run(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process. (inputToInternal output)
+      @ Out, None
+    """
+    if self.ETformat == 'OpenPSA':
+      self.outputDict = self.runOpenPSA(input)
+
+  def runOpenPSA(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process. (inputToInternal output)
+      @ Out, None
+    """
+    fileToOpen = input[0].getPath() + input[0].getFilename()
+    root = ET.parse(fileToOpen)
+    eventTree = root.findall('initial-state')
+
+    if len(eventTree) > 1:
+      self.raiseAnError(IOError,'ETimporter: more than one initial-state identified')
+
+    eventTree = eventTree[0]
+
+    ## These outcomes will be encoded as integers starting at 0
+    outcomes = []
+
+    ## These self.variables will be mapped into an array where there index
+    self.variables = []
+    values = {}
+
+    for node in root.findall('define-functional-event'):
+      event = node.get('name')
+
+      ## First, map the variable to an index by placing it in a list
+      self.variables.append(event)
+
+      ## Also, initialize the dictionary of values for this variable so we can
+      ## encode them as integers as well
+      values[event] = []
+
+      ## Iterate through the forks that use this event and gather all of the
+      ## possible states
+      for fork in self.findAllRecursive(eventTree, 'fork'):
```

Comment:
I think you can use ```Element.iter()``` method from python ET instead to create your own method here.